common/djangoapps/enrollment/api.py

```diff
@@ -225,7 +225,8 @@ def update_enrollment(user_id, course_id, mode=None, is_active=None):
         }
 
     """
-    _validate_course_mode(course_id, mode)
+    if mode is not None:
+        _validate_course_mode(course_id, mode)
```

Comment:
Just in case I'm missing something, how does this change result in "honor" becoming the default mode within the scope of this API function? As I understand it, this change prevents the API from treating `None` as an "invalid mode". `None` is a perfectly acceptable value to pass to `CourseEnrollment.update_enrollment()`; doing so causes the user's current enrollment mode to remain unchanged.